spec-conformance-tests/05-simple-values/tests/float.bal

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/test;
+import utils;
+
+// floating-point-type-descriptor := float | decimal
+// floating-point-literal := DecimalFloatingPointNumber | HexFloatingPointLiteral
+// DecimalFloatingPointNumber := DecimalNumber Exponent | DottedDecimalNumber [Exponent]
+// DottedDecimalNumber := DecimalNumber . Digit* | . Digit+
+// Exponent := ExponentIndicator [Sign] Digit+
+// ExponentIndicator := e | E
+// HexFloatingPointLiteral := HexIndicator HexFloatingPointNumber
+// HexFloatingPointNumber := HexNumber HexExponent | DottedHexNumber [HexExponent]
+// DottedHexNumber := HexDigit+ . HexDigit* | . HexDigit+
+// HexExponent := HexExponentIndicator [Sign] Digit+
+// HexExponentIndicator := p | P
+// Sign := + | -
+@test:Config {
+    dataProvider: "floatValueDataProvider"
+}
+function testFloat(float f1, float f2, float f3, float f4) {
+    test:assertEquals(f1, f2, msg = "expected float values to be equal");
+    test:assertEquals(f1, f3, msg = "expected float values to be equal");
+    test:assertEquals(f1, f4, msg = "expected float values to be equal");
+}
+
+@test:Config {
+    dataProvider: "decimalValueDataProvider"
+}
+function testDecimal(decimal d1, decimal d2, decimal d3, decimal d4) {
+    test:assertEquals(d1, d2, msg = "expected decimal values to be equal");
+    test:assertEquals(d1, d3, msg = "expected decimal values to be equal");
+    test:assertEquals(d1, d4, msg = "expected decimal values to be equal");
+}
+
+function floatValueDataProvider() returns (float[][]) {
+    return [
+        // DecimalFloatingPointNumber
+        [100e+12, 100E+12, 100e12, 100E12],
+        [100e-2, 100E-2, 1000e-3, 10E-1],
+        [0e2, 0E2, 0e-2, 0e-2],
+        [0.0, .0, 0.0, 0.00], // TODO: Calrify if 0. is valid
```

Comment:
We can remove the todos now and maybe add commented out deviation tests for `0.` right?